### PR TITLE
chore: Move dependencies to devDependencies (to reduce library weight)

### PR DESCRIPTION
**Pull request by zhibek**
#### To pass testing pipeline, these need to follow Conventional Commits spec:
https://www.conventionalcommits.org/en/v1.0.0/
e.g.
`feat: create NewForm component`
or
`fix: update broken link in NewForm component`

#### What is it doing?
Move `dependencies` that don't need to be published (i.e. supporting dependencies, those not referenced in the codebase) to `devDependencies`.

#### Why is this required?
Reducing the list of `dependencies` reduces the weight of the library, improving performance.

#### link to Jira ticket:
Add a link to the Jira ticket.

### Quick Checklist:
- [x] My PR title follows the Conventional Commit spec.

- [x] I have filled out the PR description as per the template above.

- [ ] I have added tests to cover new or changed behaviour.

- [ ] I have updated any relevant documentation.

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -3,7 +3,10 @@ module.exports = function (api) {
     api.cache(true);
 
     const presets = [ ["react-app", { "absoluteRuntime": false }] ];
-    const plugins = [ ["babel-plugin-styled-components"] ];
+    const plugins = [
+        ["babel-plugin-styled-components"],
+        ["import", { "libraryName": "lodash", "libraryDirectory": "", "camel2DashComponentName": false }]
```
Comment (AndyEPhipps) [inline babel.config.js:8]: Just so I'm clear as to what this is doing, it's overriding a lodash setting directly, so it won't collide with an ESlint setting in react-payments, or ...?

**Review by AndyEPhipps — APPROVED** [review]
Good stuff here, @zhibek - I never even thought about how all those dev dependencies come along for the ride with the CL

File: babel.config.js (hunk 1 above)
Comment (zhibek) [reply babel.config.js:8]: Hey @AndyEPhipps. I installed a babel plugin [babel-plugin-import](https://www.npmjs.com/package/babel-plugin-import) - the config you highlighted is the config for that plugin, setting it to optimise our `lodash` imports so we only import the parts of the `lodash` library we actually use (so we import 3x functions, not all ~100 functions that lodash has).

Comment (AndyEPhipps) [reply babel.config.js:8]: Ah I see, very nice!

**Pull request merged**